src/agent/testresultreader.ts

```diff
@@ -228,39 +228,19 @@ export class ResultReader {
         var testSuiteSummary = new TestSuiteSummary();
         var totalRunDuration = 0;
         var totalTestCaseDuration = 0;
-
-        if (runTitle && 0 !== runTitle.length) {
-            testSuiteSummary.name = this.fileNumberToTitle(runTitle, parseInt(fileNumber));
-        }
-        else if (rootNode.attributes().name) {
-            testSuiteSummary.name = rootNode.attributes().name;
-        }
-        else {
-            testSuiteSummary.name = this.runTitleJunit(fileName, parseInt(fileNumber));
-        }
-
-        if (rootNode.attributes().hostname) {
-            testSuiteSummary.host = rootNode.attributes().hostname;
-        }
-
-        //assume runtimes from xml are current local time since timezone information is not in the xml. If xml date > current local date, fall back to local
-        if (rootNode.attributes().timestamp) {
-            var timestampFromXml = new Date(rootNode.attributes().timestamp);
-            if (timestampFromXml < new Date()) {
-                testSuiteSummary.timeStamp = timestampFromXml;
-            }
-        }
-
-        if (rootNode.attributes().time) {
-            totalRunDuration = parseFloat(rootNode.attributes().time); //in seconds
-        }
         
-        //find test case nodes in JUnit result xml
-        var testResults = [];
-
-        for (var i = 0; i < rootNode.testcase.count(); i++) {
-            var testCaseNode = rootNode.testcase.at(i);
-
+        function extractTestResult(testCaseNode): testifm.TestResultCreateModel {
+            function pickFirst(type: string) {
+                if (testCaseNode[type].count() > 1) {
```

Comment:
I agree; I was trying to be fairly conservative and just combined what was duplicated, but I'd be glad to add another check here.